hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/SqlPlanImpl.java

```diff
@@ -196,6 +198,124 @@ public SqlResult execute(QueryId queryId, List<Object> arguments, long timeout)
         }
     }
 
+    static class CreateDataLinkPlan extends SqlPlanImpl {
+        private final boolean replace;
+        private final boolean ifNotExists;
+        private final String name;
+        private final String type;
+        private final Map<String, String> options;
+        private final PlanExecutor planExecutor;
+
+        CreateDataLinkPlan(
+                PlanKey planKey,
+                boolean replace,
+                boolean ifNotExists,
+                String name,
+                String type,
+                Map<String, String> options,
+                PlanExecutor planExecutor
+        ) {
+            super(planKey);
+
+            this.ifNotExists = ifNotExists;
+            this.replace = replace;
+            this.name = name;
+            this.type = type;
+            this.options = options;
+            this.planExecutor = planExecutor;
+        }
+
+        boolean ifNotExists() {
+            return ifNotExists;
+        }
+
+        public boolean isReplace() {
+            return replace;
+        }
+
+        public String name() {
+            return name;
+        }
+
+        public String type() {
+            return type;
+        }
+
+        public Map<String, String> options() {
+            return options;
+        }
+
+        @Override
+        public boolean isCacheable() {
+            return false;
+        }
+
+        @Override
+        public boolean producesRows() {
+            return false;
+        }
+
+        @Override
+        public void checkPermissions(SqlSecurityContext context) {
+            context.checkPermission(new SqlPermission(name, ACTION_VIEW_LINK, ACTION_CREATE_LINK));
```

Comment:
does it mean, that you need both VIEW and CREATE permission or any of them?

Reply:
https://github.com/hazelcast/hazelcast/pull/23567/commits/aa3fe8c6b9c46cebf3eb292e70b42650c146c54e

Reply:
according to design doc in git, "view" permission is not required to "create or replace" and "drop" data link